### PR TITLE
konsave: init at 2.1.2

### LINKED ISSUE #168296: Packaging request: Konsave
**Project description**
A CLI program that will let you save and apply your Linux customizations with just one command! 
It officially supports KDE Plasma but it can be used on all other desktop environments too!

**Metadata**

* homepage URL: https://pypi.org/project/Konsave/
* source URL: https://github.com/Prayag2/konsave
* license: gpl3
* platforms: linux

### PR DESCRIPTION
**Pull request by MoritzBoehme**
###### Description of changes

Init konsave at `2.1.2` at the request of @zendo and thereby closes #168296.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

TLDR: this problem is now resolved.

Original Problem:
Package can currently not be built, because konsave pins PyYaml to `5.4.1` as mentioned in following issue https://github.com/Prayag2/konsave/issues/55.
Trying to pin to that version manually did not work, because building PyYaml at `5.4.1` failed with a SyntaxError.
I will continue packaging this as soon as https://github.com/Prayag2/konsave/pull/56 is merged.

### CODE REVIEW

File: pkgs/applications/misc/konsave/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, python3 }:
+python3.pkgs.buildPythonApplication rec {
+  pname = "konsave";
+  version = "2.1.2";
```
Comment (keenanweaver) [inline pkgs/applications/misc/konsave/default.nix:4]:
```suggestion
  version = "2.2.0";
```

Bump to latest

File: pkgs/applications/misc/konsave/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = python3.pkgs.fetchPypi {
+    inherit version;
+    pname = "Konsave";
+    sha256 = "sha256-mSXkae4yq5jQBLSFSrjytRaslKhfp1D4J7aZqBJH28Y=";
```
Comment (keenanweaver) [inline pkgs/applications/misc/konsave/default.nix:9]:
```suggestion
    sha256 = "sha256-tWarqT2jFgCuSsa2NwMHRaR3/wj0khiRHidvRNMwM8M=";
```

Bump to latest

**Review by keenanweaver — CHANGES_REQUESTED** [review]

**Comment by keenanweaver** [comment]
Working for me with the 2.2.0 version.

**Review by keenanweaver — CHANGES_REQUESTED** [review]
Would you rebase on master? I'm getting merge conflicts on the all-packages.nix file. Otherwise, it builds and works great.

**Comment by MoritzBoehme** [comment]
> Would you rebase on master? I'm getting merge conflicts on the all-packages.nix file. Otherwise, it builds and works great.

Oh, I didn't see your message. I have just rebased the changes.

File: pkgs/applications/misc/konsave/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib, python3 }:
```
Comment (keenanweaver) [inline pkgs/applications/misc/konsave/default.nix:1]:
```suggestion
{ lib, python3, fetchPypi }:
```

File: pkgs/applications/misc/konsave/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "konsave";
+  version = "2.2.0";
+
+  src = python3.pkgs.fetchPypi {
+    inherit version;
+    pname = "Konsave";
+    sha256 = "sha256-tWarqT2jFgCuSsa2NwMHRaR3/wj0khiRHidvRNMwM8M=";
```
Comment (keenanweaver) [inline pkgs/applications/misc/konsave/default.nix:10]:
```suggestion
    hash = "sha256-tWarqT2jFgCuSsa2NwMHRaR3/wj0khiRHidvRNMwM8M=";
```

**Review by keenanweaver — COMMENTED** [review]
Getting this error now, for whatever reason. I think the change I suggested should resolve it.

```
       error: attribute 'fetchPypi' missing

       at /home/keenan/.cache/nixpkgs-review/pr-168475-1/nixpkgs/pkgs/applications/misc/konsave/default.nix:7:9:

            6|
            7|   src = python3.pkgs.fetchPypi {
             |         ^
            8|     inherit version;
```

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 168475` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>konsave</li>
    <li>konsave.dist</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
Looking perfect to me. Thank you very much!

File: pkgs/applications/misc/konsave/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    pname = "Konsave";
+    hash = "sha256-tWarqT2jFgCuSsa2NwMHRaR3/wj0khiRHidvRNMwM8M=";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [ setuptools-scm ];
+  propagatedBuildInputs = with python3.pkgs; [ pyyaml setuptools ];
+
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
```
Comment (Artturin) [inline pkgs/applications/misc/konsave/default.nix:18]: Changed to preCheck because it was failing in the check phase

File: pkgs/applications/misc/konsave/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    description = "Save Linux Customization";
+    maintainers = with maintainers; [ MoritzBoehme ];
+    homepage = "https://github.com/Prayag2/konsave";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
```
Comment (Artturin) [inline pkgs/applications/misc/konsave/default.nix:25]: added platforms.linux
